gcsdumpxml.c

```diff
@@ -0,0 +1,414 @@
+  {
+    fprintf(stderr, "lbm_rcv_topic_attr_create_from_xml error: %s\n", lbm_errmsg() );
+    return;
+  }
+
+  entries = lbm_rcv_topic_attr_option_size();
+  configs_option_ptr = malloc(entries * sizeof(lbm_config_option_t));
+
+  if (LBM_FAILURE == lbm_rcv_topic_attr_dump(rcv_attr_ptr, &entries, configs_option_ptr))
+  {
+    fprintf(stderr, "lbm_rcv_topic_attr_dump error: %s\n", lbm_errmsg() );
+    return;
+  }
+  dump_utility(configs_option_ptr, entries);
+}
+
+void Dump_evq_attr_xml(const char *event_queue_name)
+{
+  lbm_config_option_t *configs_option_ptr;
+  int entries;
+  lbm_event_queue_attr_t *evq_attr_ptr;
+
+  if( LBM_FAILURE == lbm_event_queue_attr_create_from_xml (&evq_attr_ptr, event_queue_name) )
+  {
+    fprintf(stderr, "lbm_event_queue_attr_create_from_xml error: %s\n", lbm_errmsg() );
+    return;
+  }
+  entries = lbm_event_queue_attr_option_size();
+  configs_option_ptr = malloc(entries * sizeof(lbm_config_option_t));
+
+  if (LBM_FAILURE == lbm_event_queue_attr_dump(evq_attr_ptr, &entries, configs_option_ptr))
+  {
+    fprintf(stderr, "lbm_event_queue_attr_dump error: %s\n", lbm_errmsg() );
+    return;
+  }
+  dump_utility(configs_option_ptr, entries);
+}
+
+
+void Dump_wrcv_attr_xml(const char* xml_context_name, const char *  pattern, int pattern_type)
+{
+  lbm_config_option_t *configs_option_ptr;
+  int entries;
+  lbm_wildcard_rcv_attr_t *wattr_ptr;
+  
+
+  if( LBM_FAILURE ==  lbm_wildcard_rcv_attr_create_from_xml (&wattr_ptr, xml_context_name, pattern, pattern_type ) )
+  {
+    fprintf(stderr, "lbm_wildcard_rcv_attr_create_from_xml error: %s\n", lbm_errmsg() );
+    return;
+  }
+  
+  entries = lbm_wildcard_rcv_attr_option_size();
+  configs_option_ptr = malloc(entries * sizeof(lbm_config_option_t));
+  
+  if( LBM_FAILURE == lbm_wildcard_rcv_attr_dump(wattr_ptr, &entries, configs_option_ptr))
+  {
+    fprintf(stderr, "lbm_wildcard_rcv_attr_dump error: %s\n", lbm_errmsg() );
+    return;
+  }
+  dump_utility(configs_option_ptr, entries);
+  lbm_wildcard_rcv_attr_delete(wattr_ptr);
+}
+
+
+void Dump_hfx_attr_xml(const char * my_hf_topic){
+  lbm_config_option_t *configs_option_ptr;
+  int entries;
+  lbm_hfx_attr_t *hfx_attr_ptr;
+  
+  if( LBM_FAILURE == lbm_hfx_attr_create_from_xml(&hfx_attr_ptr, my_hf_topic) )
+  {
+    fprintf(stderr, "lbm_hfx_attr_create error: %s\n", lbm_errmsg() );
+    return;
+  }
+
+  entries = lbm_hfx_attr_option_size();
+  configs_option_ptr = malloc(entries * sizeof(lbm_config_option_t));
+
+  if( LBM_FAILURE == lbm_hfx_attr_dump(hfx_attr_ptr, &entries, configs_option_ptr))
+  {
+    fprintf(stderr, "lbm_hfx_attr_dump error: %s\n", lbm_errmsg() );
+    return;
+  }
+  dump_utility(configs_option_ptr, entries);
+}
+
+ 
+
+int rcv_callback(lbm_rcv_t *rcv, lbm_msg_t *msg, void *clientd){
+ printf("\n Receiver callback placeholder");
+ return 0;
+}
+
+
+const char purpose[] = "Purpose: Receive messages on a single topic.";
```

Comment:
I don't think the purpose string is correct?

Reply:
I took it out since it wasn't used, thanks.